programs/uxd/src/instructions/mint_with_router.rs

```diff
@@ -0,0 +1,425 @@
+use anchor_lang::prelude::*;
+use anchor_spl::associated_token::AssociatedToken;
+use anchor_spl::token::Mint;
+use anchor_spl::token::Token;
+use anchor_spl::token::TokenAccount;
+
+use crate::error::UxdError;
+use crate::state::controller::Controller;
+use crate::state::credix_lp_depository::CredixLpDepository;
+use crate::state::identity_depository::IdentityDepository;
+use crate::state::mercurial_vault_depository::MercurialVaultDepository;
+use crate::utils::calculate_depositories_mint_collateral_amount;
+use crate::utils::calculate_depositories_mint_collateral_amount::DepositoryInfoForMintCollateralAmount;
+use crate::utils::calculate_depositories_target_redeemable_amount;
+use crate::utils::calculate_depositories_target_redeemable_amount::DepositoryInfoForTargetRedeemableAmount;
+use crate::utils::validate_collateral_amount;
+use crate::validate_is_program_frozen;
+use crate::CONTROLLER_NAMESPACE;
+use crate::CREDIX_LP_DEPOSITORY_NAMESPACE;
+use crate::CREDIX_LP_EXTERNAL_PASS_NAMESPACE;
+use crate::IDENTITY_DEPOSITORY_COLLATERAL_NAMESPACE;
+use crate::IDENTITY_DEPOSITORY_NAMESPACE;
+use crate::MERCURIAL_VAULT_DEPOSITORY_LP_TOKEN_VAULT_NAMESPACE;
+use crate::MERCURIAL_VAULT_DEPOSITORY_NAMESPACE;
+use crate::ROUTER_CREDIX_LP_DEPOSITORY_0_INDEX;
+use crate::ROUTER_IDENTITY_DEPOSITORY_INDEX;
+use crate::ROUTER_MERCURIAL_VAULT_DEPOSITORY_0_INDEX;
+
+#[derive(Accounts)]
+#[instruction(collateral_amount: u64)]
```

Comment:
Not needed, not used

Reply:
Im a bit confused, I think the collateral amount is used for computing: redeemable_circulating_supply_after

Reply:
In ```#[instruction()]``` you only need to declare the variables used in the accounts description part. It is unrelated  with `handler`.

You can see here for example, we don't declare `collateral_amount`: https://github.com/UXDProtocol/uxd-program/blob/f4175dce6a923a2a99270582dfed7c53f6671d15/programs/uxd/src/instructions/mint_with_identity_depository.rs#L18

Reply:
Ah thanks for the explaination! done!